### PR TITLE
Separate CLI client config from server config

### LINKED ISSUE #45: Separate CLI config from server config
We now have 1 type of config (core/engine) which is loaded for both CLI and server, which creates ambiguity. This should be separated; the CLI should have specific flags (e.g. remote server address, time-outs) which are specific for the invocation of CLI and the server config shouldn't include these CLI-specific flags.

### PR DESCRIPTION
**Pull request by reinkrul**
Fixes #45 

> We now have 1 type of config (core/engine) which is loaded for both CLI and server, which creates ambiguity. This should be separated; the CLI should have specific flags (e.g. remote server address, time-outs) which are specific for the invocation of CLI and the server config shouldn't include these CLI-specific flags.

Changes:

- `NutsConfig` renamed to `ServerConfig` which is only loaded when running the `server` command
- Introduced `ClientConfig`, which is only loaded when running client commands
- Introduced global `timeout` config option for `ClientConfig`, replacing the `vdr.clientTimeout` config option
- Moved normalization (prepend address with `http://` if not present)  of remote server address (client config) from VDR to `ClientConfig`
- Moved server config loading to `server` command, so it doesn't get loaded when running client commands

### CODE REVIEW

**Comment by woutslakhorst** [comment]
hmmm lots of conflicts with #51

**Review by stevenvegt — COMMENTED** [review]
Can you add the generation op the options rst to the makefile?

File: core/config.go [hunk 1]
```diff
@@ -1,158 +1,33 @@
-/*
```
Comment (stevenvegt) [inline core/config.go:1]: Why are these copyright messages removed?

**Comment by reinkrul** [comment]
> Can you add the generation op the options rst to the makefile?

They already are, that's unchanged (only expanded)

File: core/config.go (hunk 1 above)
Comment (reinkrul) [reply core/config.go:1]: Oops, renamed file than made a new one with the same name. Added them.

File: crypto/cmd/cmd.go [hunk 2]
```diff
@@ -106,17 +104,18 @@ func Cmd() *cobra.Command {
 			cmd.Println(publicKeyAsPEM)
 		},
 	})
-
+	cmd.PersistentFlags().AddFlagSet(core.NewClientConfig().FlagSet())
```
Comment (woutslakhorst) [inline crypto/cmd/cmd.go:107]: Only the crypto command adds the flagset at this point. The vdr and network do not add it here. Why the difference?

File: core/client_config.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+	}
+}
+
+// Load loads the client config from environment variables and commandline params.
+func (cfg *ClientConfig) Load() error {
+	return loadConfigIntoStruct(cfg.FlagSet(), cfg, koanf.New(defaultDelimiter))
+}
+
+// FlagSet returns the flags for configuring the client config.
+func (cfg *ClientConfig) FlagSet() *pflag.FlagSet {
```
Comment (woutslakhorst) [inline core/client_config.go:56]: doesn't need to be member func

File: README.rst [hunk 4]
```diff
@@ -159,3 +156,15 @@ the options table must be regenerated using the Makefile:
...
+CLI options
+***********
+
+The following options can be supplied when running CLI commands:
+
+=======  ==============  ===================================================================================================================================
+Key      Default         Description
+=======  ==============  ===================================================================================================================================
+address  localhost:1323  Address of the remote node. Must contain at least host and port, URL scheme may be omitted. In that case it 'http://' is prepended.
+timeout  10s             Client time-out when performing remote operations.
```
Comment (woutslakhorst) [inline README.rst:168]: my guess it uses unix style time strings, like 10s, 1m, etc? Would be good to add to description

**Review by woutslakhorst — CHANGES_REQUESTED** [review]

File: crypto/cmd/cmd.go (hunk 2 above)
Comment (reinkrul) [reply crypto/cmd/cmd.go:107]: good spot, leftover from reverted refactor

**Review by woutslakhorst — APPROVED** [review]

**Pull request merged**